src/server/api/hooks.ts

```diff
@@ -0,0 +1,229 @@
+import { APIConfig, FastifyTypebox } from './index.js'
+import { Type } from '@sinclair/typebox'
+import Store from '../store/index.js'
+import { WebHookSchema } from '../store/HookStore.js'
+import HookSystem from '../hooksystem.js'
+import ActivityPubSystem from '../apsystem.js'
+
+export const hookRoutes = (cfg: APIConfig, store: Store, hookSystem: HookSystem, apsystem: ActivityPubSystem) => async (server: FastifyTypebox): Promise<void> => {
+  // For ModerationQueued
+  server.put('/:actor/hooks/moderationqueued', {
+    schema: {
+      params: Type.Object({
+        actor: Type.String()
+      }),
+      body: WebHookSchema,
+      response: {
+        200: Type.String(),
+        409: Type.String()
```

Comment:
i don't know fastify but i got confused as to where `Reply` is used rather than `schema` for response typing

Reply:
It's a bit weird, we need both the schema and the Reply. The schema is used for the swaggger docs and runtime validation, the other is a Typescript type hint for compile time validation.